proxy/api.go

```diff
@@ -109,6 +110,67 @@ func (p *Proxy) getContent(w http.ResponseWriter, r *http.Request) {
 	return
 }
 
+// invalidateCache will invalidate all cached contentserver export files
+func (p *Proxy) invalidateCacheAll(w http.ResponseWriter, r *http.Request) {
+	// extract request data
+	workspace := strings.TrimSpace(
+		strings.ToLower(r.URL.Query().Get("workspace")),
+	)
+	user := r.Header.Get("X-User")
+
+	// validate workspace
+	if workspace == "" {
+		workspace = cms.WorkspaceLive
+	}
+
+	// logger
+	log := p.setupLogger(r, "invalidateCacheAll").WithFields(logrus.Fields{
+		logging.FieldWorkspace: workspace,
+		"user":                 user,
+	})
+
+	cachedItems, err := p.contentCache.GetAll()
+	if err != nil {
+		log.WithError(err).Error("")
+		http.Error(
+			w,
+			http.StatusText(http.StatusInternalServerError),
+			http.StatusInternalServerError,
+		)
+	}
+
+	if len(p.config.Neos.Dimensions) == 0 {
+		log.Warn("no neos dimension configured")
+	}
+
+	for _, ci := range cachedItems {
+		p.contentCache.Invalidate(ci.ID, ci.Dimension, ci.Workspace)
+
+		// load workspace worker
+		workspaceCache, workspaceOK := p.workspaceCaches[ci.Workspace]
+		if !workspaceOK {
+			log.Warn("unknown workspace")
+			http.Error(
+				w,
+				"cache invalidation failed: unknown workspace",
+				http.StatusBadRequest,
+			)
+			return
+		}
+
+		// add invalidation request to queue (contentserver export)
+		workspaceCache.Invalidate()
+	}
+
+	w.WriteHeader(http.StatusAccepted)
+	msg := fmt.Sprintf(
+		"%d cache invalidation requests accepted",
+		len(cachedItems),
+	)
+	w.Write([]byte(msg))
+	log.Debug(msg)
```

Comment:
Please use a field with (e.g. log.WithField(...) and use a constant string for msg (e.g. simply "cache invalidation requests accepted") - otherwise it's a hazzle when you want to search for that in kibana